x-pack/plugins/infra/public/hooks/use_kibana_space.ts

```diff
@@ -6,18 +6,21 @@
  */
 
 import useAsync from 'react-use/lib/useAsync';
-import type { Space } from '@kbn/spaces-plugin/public';
 import { useKibanaContextForPlugin } from './use_kibana';
 
 export type ActiveSpace =
   | { isLoading: true; error: undefined; space: undefined }
   | { isLoading: false; error: Error; space: undefined }
-  | { isLoading: false; error: undefined; space: Space };
+  | { isLoading: false; error: undefined; space: { id: string } | undefined };
 
 export const useActiveKibanaSpace = (): ActiveSpace => {
   const kibana = useKibanaContextForPlugin();
+  // Fallback to default if spaces plugin is not available
+  const getDefaultSpaceAsPromise = async () => await { id: 'default' };
```

Comment:
It's returning a promise because of the `useAsync`

Reply:
**nit:** async/await is maybe excessive, use the `Promise.resolve()` function to resolve into a promise a sync value:
```js
const getDefaultSpaceAsPromise = () => Promise.resolve({ id: 'default' });
```
also you can move it out the hook declaration as it is a function with no dependencies at all

Reply:
Thanks for the suggestion! Done 🙂